compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -459,16 +459,24 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeInterpreter<'mir,
         dest: &MPlaceTy<'tcx, Self::Provenance>,
         target: Option<mir::BasicBlock>,
         _unwind: mir::UnwindAction,
-    ) -> InterpResult<'tcx> {
+    ) -> InterpResult<'tcx, Option<ty::Instance<'tcx>>> {
         // Shared intrinsics.
         if ecx.emulate_intrinsic(instance, args, dest, target)? {
-            return Ok(());
+            return Ok(None);
         }
         let intrinsic_name = ecx.tcx.item_name(instance.def_id());
 
         // CTFE-specific intrinsics.
         let Some(ret) = target else {
-            throw_unsup_format!("intrinsic `{intrinsic_name}` is not supported at compile-time");
+            if ecx.tcx.intrinsic(instance.def_id()).unwrap().must_be_overridden {
```

Comment:
```suggestion
            // Handle diverging intrinsics.
            if ecx.tcx.intrinsic(instance.def_id()).unwrap().must_be_overridden {
```
Maybe one day we'll find a way to handle diverging and returning intrinsics with less code duplication...